lib/models/LifecycleConfiguration.js

```diff
@@ -547,6 +547,50 @@ class LifecycleConfiguration {
         return null;
     }
 
+    /**
+     * Ensure that transition rules are at least a day apart from each other.
+     * @param {object} params - Given function parameters
+     * @param {string} [params.days] - The days of the current transition
+     * @param {string} [params.date] - The date of the current transition
+     * @param {string} params.storageClass - The storage class of the current
+     * rule
+     * @param {string} params.rule - The current rule
+     * @return {undefined}
+     */
+    _checkTimeGap(params) {
+        // For testing purposes, we want to allow smaller time intervals.
+        if (process.env.CI === 'true') {
+            return undefined;
+        }
+        const { days, date, storageClass, rule } = params;
+        const invalidTransition = rule.Transition.find(transition => {
+            if (storageClass === transition.StorageClass[0]) {
+                return false;
+            }
+            if (days !== undefined) {
+                return Number.parseInt(transition.Days[0], 10) === days;
```

Comment:
`NaN` may be of a concern here.

Reply:
Okay, I will double check if I can get the lifecycle config to take with invalid entries. If so, there may be other places where we'd need to add some checks. Thanks for noting.